util/bux/translation-functions.ts

```diff
@@ -13,10 +13,136 @@ type NestedStringMap<V> = StringMap<V> | StringMap<StringMap<V>>;
 
 const writeFile = promisify(fs.writeFile);
 
+const readFileAsync = promisify(fs.readFile);
+const writeFileAsync = promisify(fs.writeFile);
+
+export async function mergeAllTranslations(platform: "web" | "desktop") {
+    const currentDirectory = process.cwd();
+    const rootDir = path.resolve(currentDirectory, "..");
+    const outputDir =
+        platform === "web"
+            ? path.join(rootDir, "web/dev-server/resources/i18n")
+            : path.join(rootDir, "desktop/resources/i18n");
+
+    // Ensure the output directory exists
+    if (!fs.existsSync(outputDir)) {
+        fs.mkdirSync(outputDir, { recursive: true });
+    }
+
+    // Define resource directories (absolute paths)
+    const resourceDirs = [
+        path.join(rootDir, "packages/common/resources/i18n/json"),
+        path.join(rootDir, "packages/playground/resources/i18n/json"),
+        path.join(rootDir, "packages/react/resources/i18n/json"),
+        path.join(rootDir, "packages/service/resources/i18n/json"),
+    ];
+
+    // Build the English file for each package before building web/desktop translations (it is a prerequisite)
+
+    const buildPackageEnglishPromises = [];
+
+    buildPackageEnglishPromises.push(
+        createEnglishTranslations(
+            path.join(rootDir, "packages/react/src/ui-react"),
+            path.join(rootDir, "packages/react/i18n"),
+            "lib.react"
+        )
+    );
+    buildPackageEnglishPromises.push(
+        createEnglishTranslations(
+            path.join(rootDir, "packages/playground/src/ui-playground"),
+            path.join(rootDir, "packages/playground/i18n"),
+            "lib.playground"
+        )
+    );
+    buildPackageEnglishPromises.push(
+        createEnglishTranslations(
+            path.join(rootDir, "packages/common/src/ui-common"),
+            path.join(rootDir, "packages/common/i18n"),
+            "lib.common"
+        )
+    );
+    buildPackageEnglishPromises.push(
+        createEnglishTranslations(
+            path.join(rootDir, "packages/service/src/ui-service"),
+            path.join(rootDir, "packages/service/i18n"),
+            "lib.service"
+        )
+    );
+
+    await Promise.all(buildPackageEnglishPromises);
+
+    // Initialize an empty object to store the merged translations
+    const mergedTranslations = Object.create(null);
```

Comment:
Switch to `const mergedTranslations: Record<string, Record<string, string>> = {};` to get Typescript type checking